crates/control/src/role_assignment.rs

```diff
@@ -355,6 +355,9 @@ fn process_role_state_machine(
                 kicking_team: Team::Opponent,
             } => return (Role::Keeper, false, None),
             _ => {}
+        };
+        if let Some(SubState::PenaltyKick) = game_controller_state.sub_state {
+            return (current_role, false, None);
```

Comment:
How does this solve the issue? Did the keeper switch his role during our test? Did you test it again?

Reply:
I checked with the replayer and yes the role changed. I am currently trying to test this in the behavior sim but #666 broke all of the files except for golden goal and demonstration so now I'm down the rabbit hole of fixing all of these files as well. Will updated when I have more info

Reply:
#811 and #812 allow the behavior simulator to now show that the keeper does in fact prepare to jump. Since the behavior simulator does not have a penalty shot direction the action jump is never carried out so this needs to be tested in the lab. However yes this is the correct fix

Reply:
On second thought, yes this allows the keeper to prepare jump, but he won't actually jump because the penaltyshotdirection is restricted to the gamephase penalty shootout. I will implement this as well

Reply:
Just tested this, works as expected now!